errors.go

```diff
@@ -591,13 +619,17 @@ func IsOAuth2(e error) bool {
 	type oe interface {
 		IsOAuth2Error() bool
 	}
-	_, ok := Unwrap(e).(oe)
-	return ok
+
+	return recursiveIsError(e, func(e error) bool {
+		_, ok := e.(oe)
+		return ok
+	})
 }
 
-// IsEOF returns true if the passed error is io.EOF
+// IsEOF returns true if the passed error is io.EOF.
+// Deprecated: Use [errors.Is] instead.
```

Comment:
There are around 20 usages in Teleport we'd have to fix. Alternatively I can drop the deprecation, or silence it on Teleport.

```
$ rg trace.IsEOF --type go -c
integrations/lib/watcherjob/watcherjob.go:1
lib/auth/grpcserver.go:2
lib/srv/sess.go:1
lib/teleterm/gatewaytest/helpers.go:1
lib/inventory/inventory.go:3
lib/sshutils/server.go:1
lib/utils/errors.go:1
api/client/inventory.go:5
api/client/client.go:1
api/client/inventory_test.go:1
api/internalutils/stream/stream.go:2
integration/integration_test.go:1
```